sql/core/src/test/scala/org/apache/spark/sql/execution/streaming/state/ListStateSuite.scala

```diff
@@ -244,4 +244,32 @@ class ListStateSuite extends StateVariableSuiteBase {
       }
     }
   }
+
+  test("ListState TTL with non-primitive types") {
```

Comment:
Is this related to this change ?

Reply:
Yes, the underlying encoding/decoding function is different with the previous approach (we were encoding the non-primitive objects into binary array and passing the binary array to the underlying layers, but we are passing unsafeRow with multiple fields), so I would like to have this suite here for checking if the encoding is working as expected for non-primitive types.

Reply:
cool - sounds good